File: C++/Greedy/구명보트.cpp
#include <iostream>
#include <algorithm>
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> people, int limit) {
    int answer = 0;
    
    int left = 0;
    int right = people.size() - 1;
    
    sort(people.begin(), people.end());
    
        
    while (true){
        if (left > right){
            break;
        }
        
            
        if(people[left] + people[right] <= limit){
            left += 1;
            right -= 1;
        } else{
            right -= 1;
        }
        
        answer += 1;
        
        if (left == right){
            answer  += 1;
            break;
        }
    }
    
    
    
    return answer;
}

File: Python/Greedy/구명보트.py
def solution(people, limit):
    ans = 0
    
    people.sort()
    sum = 0
    
    r = len(people) - 1
    l = 0
    
    while True:
        
        if r < l:
            break
        
        if people[l] + people[r] <= limit:
            l += 1
            r -= 1
            
        else:
            r -= 1
        
        ans += 1 
        
        if l == r:
            ans += 1
            break
    
    return ans
